Sources/OpenAPIRuntime/Interface/Body.swift

```diff
@@ -0,0 +1,535 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import class Foundation.NSLock
+import protocol Foundation.LocalizedError
+import struct Foundation.Data  // only for convenience initializers
+
+/// The type representing a request or response body.
+public final class Body: @unchecked Sendable {
+
+    /// The underlying data type.
+    public typealias DataType = ArraySlice<UInt8>
+
+    /// How many times the provided sequence can be iterated.
+    public enum IterationBehavior: Sendable {
+
+        /// The input sequence can only be iterated once.
+        ///
+        /// If a retry or a redirect is encountered, fail the call with
+        /// a descriptive error.
+        case single
+
+        /// The input sequence can be iterated multiple times.
+        ///
+        /// Supports retries and redirects, as a new iterator is created each
+        /// time.
+        case multiple
```

Comment:
Are we ever expecting an `AsyncSequence` based body to be `multiple` here? Right now that just doesn't exist in the ecosystem.

Reply:
Yes, for example any body created from a sync collection can be iterated multiple times. An async sequence for reading a file could also be legal to iterate multiple times. Basically, we want to provide the tools for those who want to support e.g. retryable file uploads.